### PR TITLE
add egui-material-icons & move workspace dependencies

### PR DESCRIPTION
**Pull request by bircni**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -24,3 +26,5 @@ pub use egui_suspense as suspense;
 pub use egui_thumbhash as thumbhash;
 #[cfg(feature = "virtual_list")]
 pub use egui_virtual_list as virtual_list;
+#[cfg(feature = "material_icons_proc_macro")]
+pub use material_icons_proc_macro;
```
Comment (lucasmerlin) [inline src/lib.rs:30]: I don't think anyone would want to use the proc macro directly, so I don't think we need to expose it here.

**Review by lucasmerlin — APPROVED** [review]
Thank you! Moving egui-material-icons here makes total sense. And moving shared dependencies to the root Cargo.toml is great, too.

Please split this in two PRs next time, which would make it a lot easier to review

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -14,6 +14,14 @@ jobs:
...
     runs-on: ${{ matrix.os }}
     steps:
+      - name: Install packages (Linux)
+        if: runner.os == 'Linux'
+        uses: awalsh128/cache-apt-pkgs-action@v1.4.2
+        with:
+          packages: libxcb-render0-dev libxcb-shape0-dev libxcb-xfixes0-dev libxkbcommon-dev libssl-dev libgtk-3-dev # libgtk-3-dev is used by rfd
+          version: 1.0
+          execute_install_scripts: true
+
```
Comment (lucasmerlin) [inline .github/workflows/check.yml:24]: Why is this needed?

**Comment by bircni** [comment]
Sorry for not splitting up sorry :-/

File: .github/workflows/check.yml (hunk 2 above)
Comment (bircni) [reply .github/workflows/check.yml:24]: https://github.com/lucasmerlin/hello_egui/actions/runs/11125313206/job/30912734374

this happens otherwise

Comment (bircni) [reply .github/workflows/check.yml]: btw still happens - still trying to fix

Comment (bircni) [reply .github/workflows/check.yml]: this came up as i removed the exclude from clippy

**Comment by bircni** [comment]
image

**Comment by bircni** [comment]
FYI:
Windows uses \r\n for newlines, while Linux and macOS use \n. As the include_str!("../MaterialSymbolsRounded-Regular.codepoints") file was created on a Unix-like system, the split might leave \r at the end of each line on Windows, causing issues during parsing.

thats why the new proc-macro was failing on windows

**Comment by bircni** [comment]
@lucasmerlin halleluja all ci things fixed!

**Comment by lucasmerlin** [comment]
Awesome, thank you!

**Comment by bircni** [comment]
Time for an update to crates?

**Comment by lucasmerlin** [comment]
Yes definitely, I plan to release a new version tomorrow after doing some finishing touches to egui_flex
